src/runtime/FinalizationRegistryObject.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021-present Samsung Electronics Co., Ltd
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301
+ *  USA
+ */
+
+#ifndef __EscargotFinalizationRegistryObject__
+#define __EscargotFinalizationRegistryObject__
+
+#include "runtime/Object.h"
+
+namespace Escargot {
+
+class FinalizationRegistryObjectItem : public gc {
+public:
+    void* operator new(size_t size);
+    void* operator new[](size_t size) = delete;
+    Value getUnregisterToken();
+    explicit FinalizationRegistryObjectItem(ExecutionState& state, Value weakRefTarget, Value heldValue, Value unregisterToken);
+
+private:
+    Value m_weakRefTarget;
+    Value m_heldValue;
+    Value m_unregisterToken;
+};
+
+class FinalizationRegistryObject : public Object {
+public:
+    typedef TightVector<FinalizationRegistryObjectItem*, GCUtil::gc_malloc_allocator<FinalizationRegistryObjectItem*>> FinalizationRegistryObjectCells;
```

Comment:
`TightVector` is usually allocated when insert/delete operations are rarely occur.
For `FinalizationRegistryObjectCells` case, I think that its better to use `Vector` structure instead.

Reply:
Would you please check the Vector type once again (changing it to `Vector`)?